mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -517,69 +516,75 @@ class WidgetAdapter(
 
             if (item.isOfTypeOrGroupType(Item.Type.Color)) {
                 val brightness = state.asBrightness
                 if (brightness != null) {
-                    seekBar.max = 100
-                    seekBar.progress = brightness
+                    seekBar.valueFrom = 0F
+                    seekBar.valueTo = 100F
+                    seekBar.value = brightness.toFloat()
                 }
             } else {
                 val number = state.asNumber
                 if (number != null) {
-                    val progress = (number.value - widget.minValue) / widget.step
-                    seekBar.progress = Math.round(progress)
+                    var progress = number.value
+                    progress = min(progress, seekBar.valueTo)
+                    progress = max(progress, seekBar.valueFrom)
+                    seekBar.value = progress
                 }
             }
         }
 
         override fun handleRowClick() {
-            if (boundWidget?.switchSupport == true) {
-                connection.httpClient.sendItemCommand(boundWidget?.item,
-                    if (seekBar.progress == 0) "ON" else "OFF")
+            val widget = boundWidget ?: return
+            if (widget.switchSupport) {
+                connection.httpClient.sendItemCommand(widget.item,
+                    if (seekBar.value <= widget.minValue) "ON" else "OFF")
             }
         }
 
-        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+        override fun onStartTrackingTouch(slider: Slider) {
             // no-op
         }
 
-        override fun onStartTrackingTouch(seekBar: SeekBar) {
-            Log.d(TAG, "onStartTrackingTouch position = ${seekBar.progress}")
+        override fun onStopTrackingTouch(slider: Slider) {
+            val value = slider.value
+            Log.e(TAG, "onValueChange value = $value")
+            val item = boundWidget?.item ?: return
+            if (item.isOfTypeOrGroupType(Item.Type.Color)) {
+                connection.httpClient.sendItemCommand(item, value.toString())
+            } else {
+                connection.httpClient.sendItemUpdate(item, item.state?.asNumber.withValue(value))
+            }
         }
 
-        override fun onStopTrackingTouch(seekBar: SeekBar) {
-            val progress = seekBar.progress
-            Log.d(TAG, "onStopTrackingTouch position = $progress")
-            val widget = boundWidget
-            val item = widget?.item ?: return
-            if (item.isOfTypeOrGroupType(Item.Type.Color)) {
-                connection.httpClient.sendItemCommand(item, progress.toString())
+        override fun getFormattedValue(value: Float): String {
+            val item = boundWidget?.item ?: return ""
+            val realValue = if (boundWidget?.step == 1F) value.toInt().toFloat() else value
+            return if (item.isOfTypeOrGroupType(Item.Type.Color)) {
+                realValue.toString()
```

Comment:
Would it be better to convert to percent here?

Reply:
I just tested with a color bulb and it works fine.

Reply:
Yes, makes sense.